README.md
Files used during testing python programs for raspibot control


history.txt
    1  ifconfig
    2  cd /etc
    3  ls
    4  cd network
    5  ls
    6  file *
    7  vi interfaces
    8  cd ..
    9  ls -l mo*
   10  lsmod
   11  sudo modprob w1-gpio
   12  sudo modprobe w1-gpio
   13  sudo modprobe w1-therm
   14  ls /sys/bus/w1
   15  ls /sys/bus/w1/devices/
   16  ls /sys/bus/w1/devices/w1_bus_master1
   17  cd network
   18  ls -ltr
   19  vi interfaces
   20  ls -l
   21  sudo vi interfaces 
   22  sudo shutdown -r now
   23  sudo ifconfig -a
   24  startx
   25  sudo shutdown -h now
   26  ls
   27  vi blink.py
   28  python blink.py 
   29  sudo python blink.py 
   30  vi temptest.py
   31  lsmod
   32  sudo modprobe w1-gpio
   33  sudo modprobe w1-therm
   34  lsmod
   35  cd /etc
   36  ls -l mo*
   37  sudo vi modules 
   38  cd
   39  python temptest.py 
   40  date;python temptest.py 
   41  vi adctest.py
   42  python adctest.py 
   43  sudo python adctest.py 
   44  clear
   45  date;python temptest.py 
   46  md raspibot
   47  mkdir raspibot
   48  ls
   49  mv *.py raspibot/
   50  touch README.md
   51  git init
   52  pwd
   53  ls
   54  mv README.md raspibot/
   55  ls a-l
   56  ls -al
   57  em .git
   58  rm .git
   59  rmdir -rf .git
   60  rmdir -f .git
   61  rmdir .git
   62  ls .git
   63  ls -al
   64  cd .git
   65  ls -l
   66  rm *
   67  ls -l
   68  rmdir branches/
   69  rmdir hooks
   70  rm hooks/*
   71  rmdir hooks
   72  rmdir info
   73  rm info/*
   74  rmdir info
   75  rmdir objects/
   76  rm objects/*
   77  cd objects/
   78  ls
   79  rmdir info
   80  rmdir pack
   81  cd ..
   82  rmdir objects/
   83  ls
   84  rm refs
   85  rmdir refs
   86  cd refs
   87  ls
   88  rmdir heads
   89  rmdir tags
   90  cd ..
   91  rmdir refs
   92  cd ..
   93  rmdir .git
   94  cd raspibot/
   95  init git
   96  git init
   97  git add README.md
   98  ls
   99  git add *,py
  100  git add *.py
  101  ls -l
  102  git commit -m "first commit"
  103  git config --global user.email randy@greigfamily.ca
  104  git config --global user.name "Randy Greig"
  105  git commit -m "first commit"
  106  #!/usr/bin/env python
  107  import time
  108  import os
  109  import RPi.GPIO as GPIO
  110  GPIO.setmode(GPIO.BCM)
  111  DEBUG = 1
  112  LOGGER = 1
  113  # read SPI data from MCP3008 chip, 8 possible adc's (0 thru 7)
  114  def readadc(adcnum, clockpin, mosipin, misopin, cspin):
  115  git remote add origin https://github.com/randygreig/rasibot.git
  116  git push -u origin master
  117  man git
  118  git remote set-url origin #!/usr/bin/env python
  119  import time
  120  import os
  121  import RPi.GPIO as GPIO
  122  GPIO.setmode(GPIO.BCM)
  123  DEBUG = 1
  124  LOGGER = 1
  125  # read SPI data from MCP3008 chip, 8 possible adc's (0 thru 7)
  126  def readadc(adcnum, clockpin, mosipin, misopin, cspin):
  127  git remote set-url origin https://github.com/randygreig/raspibot.git
  128  git push -u origin master
  129  cd ..
  130  ls
  131  ls -al
  132  ssh-keygen -t rsa -C "randy@greigfamily.ca"
  133  copy .ssh/id_rsa.pub 
  134  cat .ssh/id_rsa.pub 
  135  ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC8mCreIG3H5ZvM76NgoarmKHJF0kgR/7lr/PSPV/NqcmaB+uoO6/7DIGWdoQnwjfp4KuTTi7pcfcRdMEZv3uXEjLEbscE/AvktzOVJa91BSRtLIv7wCr313hM+ImmXWm2K5PUHVCXhd1INaRpQ5ClS/ijW+JiHdFm4sLbiOIT8aKFT3GrH++awzgRp09ynsQgskKxR7TVIf0lWPXuf3ZDejlCAxgs2zJqOEv+awJGTzM1GWguCtpvlJzD74gcUdREtWCz+9/GQ6Jkl5/mT8WUw32Rp5nUE53nTt37NDzuHTtCWbuAFMF1kwHwz8DDk4HDqwU14PphEsJ2iPwhL5+Yr randy@greigfamily.ca
  136  ssh -T git@github.com
  137  git push -u origin master
  138  cd raspibot/
  139  git push -u origin master
  140  git push origin master
  141  ls -la
  142  date
  143  git commit -m 'second commit'
  144  vi README.md 
  145  git commit -m 'second commit'
  146  git commit -a
  147  git commit -m 'second commit'
  148  git push origin master
  149  git remote set-url origin git@github.com/randygreig/raspibot.git
  150  git push origin master
  151  git remote set-url origin git@github.com:randygreig/raspibot.git
  152  git push origin master
  153  history
  154  git config --list
  155  history
  156  git status
  157  clear
  158  date;python temptest.py 
  159  sudo shutdown -h now
  160  ls
  161  cd raspibot/
  162  ls
  163  sudo python blink.py 
  164  date;python temptest.py 
  165  clear
  166  date;python temptest.py 
  167  sudo vi /etc/ntp.conf
  168  sudo pylint temptest.py 
  169  sudo pep8 temptest.py 
  170  date;python temptest.py 
  171  git status
  172  ls -ltr
  173  cat history.txt 
  174  ls
  175  git fetch
  176  ls -l
  177  git fetch
  178  ls -l
  179  git fetch --all
  180  ls
  181  ls -al
  182  git /?
  183  git --help
  184  git fetch --help
  185  git merge
  186  got checkout
  187  git checkout
  188  git status
  189  ls
  190  git merge --help
  191  git pull
  192  git pull --help
  193  ls -ltr
  194  ls -ltra
  195  ls .git
  196  ls -ltr .git
  197  ls -ltr .git/FETCH_HEAD 
  198  cat .git/FETCH_HEAD 
  199  cat .git/index 
  200  ls -ltr .git/FETCH_HEAD 
  201  ls -ltr .git/
  202  ls -ltr .git/objects/
  203  ls -ltr .git/objects/65
  204  ls -ltr .git/objects/a0
  205  ls -ltr .git/objects/cb
  206  cat .git/objects/65
  207  ls -ltr .git/objects/65
  208  cat .git/objects/65/eee98e5dfccd1ed90c2162c7ebc204c3a575f7 
  209  PuTTYPuTTYPuTTYPuTTYPuTTYPuTTY
  210  clear
  211  git remote -v
  212  git fetch origin
  213  ls -l
  214  git merge origin
  215  git merge
  216  git merge --help
  217  git merge recursive theirs
  218  git merge --help
  219  git merge -Xrecursive theirs
  220  git merge -Xrecursive
  221  ls a-l
  222  ls -al
  223  ls .git
  224  more .git/config
  225  git config --help
  226  git config merge.todefaultupstream
  227  more .git/config
  228  git config --help
  229  git config merge.defaultToUpstream
  230  more .git/config
  231  git config merge.defaultToUpstream True
  232  more .git/config
  233  git merge
  234  git clone
  235  git --help
  236  git list
  237  git branch

adctest.py
#!/usr/bin/env python
import time
import os
import RPi.GPIO as GPIO
 
GPIO.setmode(GPIO.BCM)
DEBUG = 1
LOGGER = 1
 
# read SPI data from MCP3008 chip, 8 possible adc's (0 thru 7)
def readadc(adcnum, clockpin, mosipin, misopin, cspin):
        if ((adcnum > 7) or (adcnum < 0)):
                return -1
        GPIO.output(cspin, True)
 
        GPIO.output(clockpin, False)  # start clock low
        GPIO.output(cspin, False)     # bring CS low
 
        commandout = adcnum
        commandout |= 0x18  # start bit + single-ended bit
        commandout <<= 3    # we only need to send 5 bits here
        for i in range(5):
                if (commandout & 0x80):
                        GPIO.output(mosipin, True)
                else:   
                        GPIO.output(mosipin, False)
                commandout <<= 1
                GPIO.output(clockpin, True)
                GPIO.output(clockpin, False)
 
        adcout = 0
        # read in one empty bit, one null bit and 10 ADC bits
        for i in range(12):
                GPIO.output(clockpin, True)
                GPIO.output(clockpin, False)
                adcout <<= 1
                if (GPIO.input(misopin)):
                        adcout |= 0x1
 
        GPIO.output(cspin, True)
 
        adcout /= 2       # first bit is 'null' so drop it
        return adcout
 
# change these as desired - they're the pins connected from the
# SPI port on the ADC to the Cobbler
SPICLK = 18
SPIMISO = 23
SPIMOSI = 24
SPICS = 25
 
# set up the SPI interface pins
GPIO.setup(SPIMOSI, GPIO.OUT)
GPIO.setup(SPIMISO, GPIO.IN)
GPIO.setup(SPICLK, GPIO.OUT)
GPIO.setup(SPICS, GPIO.OUT)
 
# temperature sensor connected channel 0 of mcp3008
adcnum = 0
 
while True:
        # read the analog pin 0
        read_adc0 = readadc(0, SPICLK, SPIMOSI, SPIMISO, SPICS)
 
        # convert analog reading to millivolts = ADC * ( 5000 / 1024 )
        millivolts = read_adc0 * ( 5000.0 / 1024.0)
 
        # remove decimal point from millivolts
        millivolts = "%d" % millivolts
 
        print("read_adc0:", millivolts)
 
        # read the analog pin 1
        read_adc1 = readadc(1, SPICLK, SPIMOSI, SPIMISO, SPICS)
 
        # convert analog reading to millivolts = ADC * ( 5000 / 1024 )
        millivolts = read_adc1 * ( 5000.0 / 1024.0)
 
        # remove decimal point from millivolts
        millivolts = "%d" % millivolts
 
        print("read_adc1:", millivolts)
 
        # read the analog pin  2
        read_adc2 = readadc(2, SPICLK, SPIMOSI, SPIMISO, SPICS)
 
        # convert analog reading to millivolts = ADC * ( 5000 / 1024 )
        millivolts = read_adc2 * ( 5000.0 / 1024.0)
 
        # remove decimal point from millivolts
        millivolts = "%d" % millivolts
 
        print("read_adc2:", millivolts)
 
        # read the analog pin 3
        read_adc3 = readadc(3, SPICLK, SPIMOSI, SPIMISO, SPICS)
 
        # convert analog reading to millivolts = ADC * ( 5000 / 1024 )
        millivolts = read_adc3 * ( 5000.0 / 1024.0)
 
        # remove decimal point from millivolts
        millivolts = "%d" % millivolts
 
        print("read_adc3:", millivolts)
 
        # read the analog pin 4
        read_adc4 = readadc(4, SPICLK, SPIMOSI, SPIMISO, SPICS)
 
        # convert analog reading to millivolts = ADC * ( 5000 / 1024 )
        millivolts = read_adc4 * ( 5000.0 / 1024.0)
 
        # remove decimal point from millivolts
        millivolts = "%d" % millivolts
 
        print("read_adc4:", millivolts)
        print("\n")
        print("\n")
        print("\n")
 
        time.sleep(5)




temptest.py
#!/usr/bin/python

MIN_TEMP = -100
temp1 = "/sys/bus/w1/devices/28-00000379fea9/w1_slave"
temp2 = "/sys/bus/w1/devices/28-00000447416d/w1_slave"

def read_temp1():
  # open/read/close the file with the temperature
  temp1file = open(temp1)
  text = temp1file.read()
  temp1file.close()

  # split the two lines
  lines = text.split("\n")

  # make sure the crc is valid
  if lines[0].find("YES") > 0:
    #get the 9th (10th) chunk of text and lose the t= bit
    temp = float((lines[1].split(" ")[9])[2:])
    # add a decimal point
    temp /= 1000
    temp = round(temp,2)
    return temp
  return MIN_TEMP-1

def read_temp2():
  # open/read/close the file with the temperature
  temp2file = open(temp2)
  text = temp2file.read()
  temp2file.close()

  # split the two lines
  lines = text.split("\n")

  # make sure the crc is valid
  if lines[0].find("YES") > 0:
    #get the 9th (10th) chunk of text and lose the t= bit
    temp = float((lines[1].split(" ")[9])[2:])
    # add a decimal point
    temp /= 1000
    temp = round(temp,2)
    return temp
  return MIN_TEMP-1

print(read_temp1())
print(read_temp2())




raspibot_server.py
#!/usr/bin/python

import RPi.GPIO as GPIO
import socket
import time

GPIO.setmode(GPIO.BCM)
GPIO.setwarnings(False)
GPIO.setup(7, GPIO.OUT)
GPIO.setup(8, GPIO.OUT)
GPIO.setup(9, GPIO.OUT)

robotsock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
robotsock.bind(('', 9000))
robotsock.listen(1)
print "Listening on TCP 9000"

print "Waiting for connection..."
connection, addr = robotsock.accept()
connection.setblocking(0)
print "Connected by", addr[0]

while(1):
    GPIO.output(7, False)
    GPIO.output(8, False)
    GPIO.output(9, False)
    while(1):
        try:
            sockdata = connection.recv(1)
            break
        except:
            pass
    if sockdata == 'R':
        print "red"
        GPIO.output(7, True)
        time.sleep(2)
    if sockdata == 'G':
        print "green"
        GPIO.output(8, True)
        time.sleep(2)
    if sockdata == 'B':
        print "blue"
        GPIO.output(9, True)
        time.sleep(2)
    if sockdata == 'Q':
        break

GPIO.output(7, False)
GPIO.output(8, False)
GPIO.output(9, False)

GPIO.cleanup()



raspibot-control.py
#!/usr/bin/env python

###############################################################################
# Module:   raspibot-control.py
# Created:  2013-12-28
# Author:   Randy Greig
# Version:  0.1
'''

Dependencies:
  pygame   - http://www.pygame.org/
'''
###############################################################################

print "\n================================================"
print "        raspibot Joystick control"
print "================================================"

# Import dependent Python modules
import socket
try:
	import pygame.joystick
except:
	print "\nPlease install the 'pygame' module <http://www.pygame.org/>.\n"
	quit()

robotsock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
robotsock.connect(("192.168.1.52", 9000))

# Allow for multiple joysticks
joy = []

# Handle joystick event
def handleJoyEvent(e):
    # Identify joystick axes and assign events
    if e.type == pygame.JOYAXISMOTION:
        axis = "unknown"
        if (e.dict['axis'] == 0):
            axis = "X"
        if (e.dict['axis'] == 1):
            axis = "Y"
        if (e.dict['axis'] == 2):
            axis = "Throttle"
        if (e.dict['axis'] == 3):
            axis = "Z"

        # Convert joystick value to servo position for each axis
        if (axis != "unknown"):
            # str = "Axis: %s; Value: %f" % (axis, e.dict['value'])
            pos = e.dict['value']
            move = round(pos * 90, 0)
            serv = int(90 + move)
            str = "Axis: %s; Value: %f" % (axis, serv)
            # Uncomment to display axis values:
            output(str, e.dict['joy'])

            # X Axis
            if (axis == "X"):
                pos = e.dict['value']
                # convert joystick position to servo increment, 0-180
                move = round(pos * 90, 0)
                serv = int(90 + move)
                # and send to Arduino over serial connection
                #servo.move(1, serv)
            # Y Axis
            if (axis == "Y"):
                pos = e.dict['value']
                move = round(pos * 90, 0)
                serv = int(90 + move)
                #servo.move(2, serv)
            # Z Axis
            if (axis == "Z"):
                pos = e.dict['value']
                move = round(pos * 90, 0)
                serv = int(90 + move)
                #servo.move(3, serv)
            # Throttle
            if (axis == "Throttle"):
                pos = e.dict['value']
                move = round(pos * 90, 0)
                serv = int(90 + move)
                #servo.move(4, serv)

    # Assign actions for Button DOWN events
    elif e.type == pygame.JOYBUTTONDOWN:
        # Button 1 (trigger)
        if (e.dict['button'] == 0):
            print "Trigger Down"
            robotsock.send('R')
        # Button 2
        if (e.dict['button'] == 1):
            print "Button 2 Down"
            robotsock.send('G')
        # Button 3
        if (e.dict['button'] == 2):
            print "Button 3 Down"
            robotsock.send('B')
        # Button 4
        if (e.dict['button'] == 3):
            print "Button 4 Down"
        # Button 5
        if (e.dict['button'] == 4):
            print "Button 5 Down"
        # Button 6
        if (e.dict['button'] == 5):
            print "Button 6 Down"
            robotsock.send('Q')
            quit()

    # Assign actions for Button UP events
    elif e.type == pygame.JOYBUTTONUP:
        # Button 1 (trigger)
        if (e.dict['button'] == 0):
            print "Trigger Up"
            # Set pin 13 LED to LOW for digital on/off demo
            #servo.move(99, 0)
        # Button 2
        if (e.dict['button'] == 1):
            print "Button 2 Up"
        # Button 3
        if (e.dict['button'] == 2):
            print "Button 3 Up"
        # Button 4
        if (e.dict['button'] == 3):
            print "Button 4 Up"
        # Button 5
        if (e.dict['button'] == 4):
            print "Button 5 Up"
        # Button 6
        if (e.dict['button'] == 5):
            print "Button 6 Up"

    else:
        pass

# Print the joystick position
def output(line, stick):
    print "Joystick: %d; %s" % (stick, line)

# Wait for joystick input
def joystickControl():
    while True:
        e = pygame.event.wait()
        if (e.type == pygame.JOYAXISMOTION or e.type == pygame.JOYBUTTONDOWN or e.type == pygame.JOYBUTTONUP):
            handleJoyEvent(e)

# Main method
def main():
    # Initialize pygame
    pygame.joystick.init()
    pygame.display.init()
    if not pygame.joystick.get_count():
        print "\nPlease connect a joystick and run again.\n"
        quit()
    print "\n%d joystick(s) detected." % pygame.joystick.get_count()
    for i in range(pygame.joystick.get_count()):
        myjoy = pygame.joystick.Joystick(i)
        myjoy.init()
        joy.append(myjoy)
        print "Joystick %d: " % (i) + joy[i].get_name()
    print "Depress joystick button 6 to quit.\n"

    # Run joystick listener loop
    joystickControl()

# Allow use as a module or standalone script
if __name__ == "__main__":
    main()


joytest.py
#!/usr/bin/env python

###############################################################################
# Module:   joytest.py
# Created:  2013-12-25
# Author:   Randy Greig
# Version:  0.1
'''

Dependencies:
  pygame   - http://www.pygame.org/
'''
###############################################################################

print "\n================================================"
print "        Python Joystick test"
print "================================================"

# Import dependent Python modules
try:
	import pygame.joystick
except:
	print "\nPlease install the 'pygame' module <http://www.pygame.org/>.\n"
	quit()

# Allow for multiple joysticks
joy = []

# Handle joystick event
def handleJoyEvent(e):
    # Identify joystick axes and assign events
    if e.type == pygame.JOYAXISMOTION:
        axis = "unknown"
        if (e.dict['axis'] == 0):
            axis = "X"
        if (e.dict['axis'] == 1):
            axis = "Y"
        if (e.dict['axis'] == 2):
            axis = "Throttle"
        if (e.dict['axis'] == 3):
            axis = "Z"

        # Convert joystick value to servo position for each axis
        if (axis != "unknown"):
            str = "Axis: %s; Value: %f" % (axis, e.dict['value'])
            # Uncomment to display axis values:
            output(str, e.dict['joy'])

            # X Axis
            if (axis == "X"):
                pos = e.dict['value']
                # convert joystick position to servo increment, 0-180
                move = round(pos * 90, 0)
                serv = int(90 + move)
                # and send to Arduino over serial connection
                #servo.move(1, serv)
            # Y Axis
            if (axis == "Y"):
                pos = e.dict['value']
                move = round(pos * 90, 0)
                serv = int(90 + move)
                #servo.move(2, serv)
            # Z Axis
            if (axis == "Z"):
                pos = e.dict['value']
                move = round(pos * 90, 0)
                serv = int(90 + move)
                #servo.move(3, serv)
            # Throttle
            if (axis == "Throttle"):
                pos = e.dict['value']
                move = round(pos * 90, 0)
                serv = int(90 + move)
                #servo.move(4, serv)

    # Assign actions for Button DOWN events
    elif e.type == pygame.JOYBUTTONDOWN:
        # Button 1 (trigger)
        if (e.dict['button'] == 0):
            print "Trigger Down"
            # Set pin 13 LED to HIGH for digital on/off demo
            #servo.move(99, 180)
        # Button 2
        if (e.dict['button'] == 1):
            print "Button 2 Down"
        # Button 3
        if (e.dict['button'] == 2):
            print "Button 3 Down"
        # Button 4
        if (e.dict['button'] == 3):
            print "Button 4 Down"
        # Button 5
        if (e.dict['button'] == 4):
            print "Button 5 Down"
        # Button 6
        if (e.dict['button'] == 5):
            print "Button 6 Down"
            quit()

    # Assign actions for Button UP events
    elif e.type == pygame.JOYBUTTONUP:
        # Button 1 (trigger)
        if (e.dict['button'] == 0):
            print "Trigger Up"
            # Set pin 13 LED to LOW for digital on/off demo
            #servo.move(99, 0)
        # Button 2
        if (e.dict['button'] == 1):
            print "Button 2 Up"
        # Button 3
        if (e.dict['button'] == 2):
            print "Button 3 Up"
        # Button 4
        if (e.dict['button'] == 3):
            print "Button 4 Up"
        # Button 5
        if (e.dict['button'] == 4):
            print "Button 5 Up"
        # Button 6
        if (e.dict['button'] == 5):
            print "Button 6 Up"

    # Assign actions for Coolie Hat Switch events
    elif e.type == pygame.JOYHATMOTION:
        if (e.dict['value'][0] == -1):
            print "Hat Left"
            #servo.move(4, 0)
        if (e.dict['value'][0] == 1):
            print "Hat Right"
            #servo.move(4, 180)
        if (e.dict['value'][1] == -1):
            print "Hat Down"
        if (e.dict['value'][1] == 1):
            print "Hat Up"
        if (e.dict['value'][0] == 0 and e.dict['value'][1] == 0):
            print "Hat Centered"
            #servo.move(4, 90)
		
    else:
        pass

# Print the joystick position
def output(line, stick):
    print "Joystick: %d; %s" % (stick, line)

# Wait for joystick input
def joystickControl():
    while True:
        e = pygame.event.wait()
        if (e.type == pygame.JOYAXISMOTION or e.type == pygame.JOYBUTTONDOWN or e.type == pygame.JOYBUTTONUP or e.type == pygame.JOYHATMOTION):
            handleJoyEvent(e)

# Main method
def main():
    # Initialize pygame
    pygame.joystick.init()
    pygame.display.init()
    if not pygame.joystick.get_count():
        print "\nPlease connect a joystick and run again.\n"
        quit()
    print "\n%d joystick(s) detected." % pygame.joystick.get_count()
    for i in range(pygame.joystick.get_count()):
        myjoy = pygame.joystick.Joystick(i)
        myjoy.init()
        joy.append(myjoy)
        print "Joystick %d: " % (i) + joy[i].get_name()
    print "Depress joystick button 6 to quit.\n"

    # Run joystick listener loop
    joystickControl()

# Allow use as a module or standalone script
if __name__ == "__main__":
    main()
